nix/host/scheduled-tasks.nix

```diff
@@ -1,23 +1,95 @@
-{ lib, flake, ... }: {
-  systemd = {
-    timers.foo = {
-      description = "Run task-foo every 5 minutes";
-      wantedBy = [ "timers.target" ];
-      timerConfig = {
-        OnBootSec = "5m";
-        OnUnitActiveSec = "5m";
-        Unit = "foo.service";
+# TODO make this a NixOS module
+{ lib, flake, config, ... }:
+let
+  networks = [ "preview" "preprod" "mainnet" ];
+  # TODO make this parametric with mode data source
+  basePaths = [ "kupo-data" "exported-snapshots/kupo" "mithril-snapshots" ];
+  mkLocalDataPathFromBase = basePath: lib.concatStringsSep " " (map (network: "${basePath}/${network}") networks);
+  localDataPaths = lib.concatStringsSep " " (map mkLocalDataPathFromBase basePaths);
+  # TODO make this parametric with mode data source
+  mkS3DataPath = network: "kupo/${network}/";
+  s3DataPaths = lib.concatStringsSep " " (map mkS3DataPath networks);
+  r2Endpoint = "https://5c90369860b916812808cd543a1d782b.r2.cloudflarestorage.com";
+  bucketName = "cardanow";
+  mkCardanowService = network: {
+    systemd = {
+      timers."cardanow-${network}" = {
+        description = "Run cardanow for ${network} every 24 hours";
+        wantedBy = [ "timers.target" ];
+        timerConfig = {
+          OnBootSec = "0m";
+          OnUnitActiveSec = "48h";
```

Comment:
This means that the timer triggers every 48h and the first time it triggers as soon as the machine is booted, perhaps it would be better just specifying a timestamp (e.g. midnight every two days).
This way we could also keep timers for different networks out of phase.

I'm not saying to address this in this PR, just wanted to know what you think and in case make you create a TODO comment.

Reply:
Added a comment for this